src/query/functions/temporal/rate.go

```diff
@@ -36,36 +36,154 @@ const (
 	// IDeltaType calculates the difference between the last two values in the time series.
 	// IDeltaTemporalType should only be used with gauges.
 	IDeltaType = "idelta"
+
+	// RateType calculates the per-second average rate of increase of the time series.
+	RateType = "rate"
+
+	// DeltaType calculates the difference between the first and last value of each time series.
+	DeltaType = "delta"
+
+	// IncreaseType calculates the increase in the time series.
+	IncreaseType = "increase"
 )
 
 // NewRateOp creates a new base temporal transform for rate functions
 func NewRateOp(args []interface{}, optype string) (transform.Params, error) {
-	if optype == IRateType || optype == IDeltaType {
+	if optype == IRateType || optype == IDeltaType || optype == RateType ||
+		optype == IncreaseType || optype == DeltaType {
 		return newBaseOp(args, optype, newRateNode, nil)
 	}
 
 	return nil, fmt.Errorf("unknown rate type: %s", optype)
 }
 
 func newRateNode(op baseOp, controller *transform.Controller, opts transform.Options) Processor {
-	isRate := op.operatorType == IRateType
+	isRate := op.operatorType == IRateType || op.operatorType == RateType
```

Comment:
Doing a bunch of comparisons here that you've already done; might be better to refactor to something like this
```golang
var isRate, isCounter bool
switch(op.operatorType)
case IRateType:
 isRate = true
 rateFunc = irateFunc
```